### ISSUE TITLE
Training data cannot span multiple startdate/enddate windows, and median_per_id merges them silently

### ISSUE THREAD

**Issue by nikhilsrajan** [nikhilsrajan/fsd#84]
**`create_training_data` cannot produce training data spanning more than one `startdate`/`enddate` window, and the failure is silent rather than loud.** Raised by the user 2026-08-21 while recovering the rationale for `overwrite_setup_csv` (spec 50 §7 Q3): the original `setup` appended to `input.csv` specifically so it could be run several times with different windows, accumulating units — *"that allowed for creation of training data with different start and end dates"*. That capability was designed for, is half-built, and is currently unreachable.

**Why it is unreachable today.** `run_create_datacube` deletes `input.csv` on every call (`overwrite_setup_csv=True`, inherited from `fetch_satdata` — spec 50 §1), so nothing accumulates. Spec 50 removes the delete and restores append-plus-dedupe, at which point multi-window rows *can* coexist in one `input.csv` — and then the three problems below become reachable. **They should be fixed before or with that change, not after.**

**1. `ids` have no window component, so the windows collide.** `_UNIT_IDENTITY_COLS` correctly distinguishes rows by `(id, startdate, enddate, bands, mosaic_days, mosaic_scheme, scl_mask_classes)`, so two windows for field `123` are two legitimate rows. But `flatten` writes `ids.npy` from the `id` column alone, so both land as `123` with nothing to tell them apart. A modeller reading the arrays cannot recover which window a sample came from.

**2. `median_per_id` then merges the windows together, silently.** `fsd/model/features.py::median_per_id` does `np.unique(ids)` and takes `np.nanmedian` across every row sharing an id:

```python
unique_ids, inverse = np.unique(ids, return_inverse=True)
for i in range(n_ids):
    rows = inverse == i
    data_median[i] = np.nanmedian(data[rows], axis=0)
```

So a field observed in Apr–Sep 2018 *and* Apr–Sep 2019 does not become two training samples — it becomes **one sample that is the median of both years**. That is not a crash, a warning, or a shape mismatch; it is a quietly wrong training set, and `aggregate="median_per_id"` is what the Austria demo uses. This is the part that makes the issue urgent rather than merely missing.

**3. Only equal-length windows can stack at all.** `flatten` requires a shared `timestamps` axis (`CLAUDE.md`: cubes over the same start/end/mosaic_days share an identical axis, `T = ceil((end-start)/mosaic_days)`). Apr–Sep 2018 and Apr–Sep 2019 at `mosaic_days=20` are both `T=10` and stack fine — which is exactly the interesting case, multi-year samples of the same season. Apr–Sep and Apr–Jun do not, and should be refused with a message naming both `T`s rather than failing somewhere inside a reshape.

**What a fix looks like** (sketch, not a decision — this wants its own spec):

- give the flattened arrays a **window component** alongside `ids` — either a compound id (`123@20180401_20180930_m20`) or a parallel `windows.npy`, so a sample is addressable by (field, window);
- make `median_per_id` group by **the unit identity, not the bare id** — the docstring's premise, *"a field's pixels share a label"*, is true within a window and false across them;
- preflight the `T` agreement across the rows being flattened, and refuse with both numbers on a mismatch (spec 47 Part D: caller misuse raises);
- decide whether `label_col` may legitimately differ per window for the same field (a parcel's crop changes between years — which is the whole point of multi-year training data, and means labels must be per (field, window) too).

**Related:** spec 50 (removes the delete that currently hides this), spec 38 D13 / `_dedupe_on_unit_identity` (already distinguishes rows correctly — the row layer is right, the array layer is not), spec 39 (flatten), #83.

**Area:** api/create_training_data, datacube/flatten, model/features

**Why deferred / note:** not currently reachable, because the `input.csv` delete prevents accumulation — so no existing run is affected and nothing is silently wrong today. It becomes reachable the moment spec 50 lands, which is why it is filed now rather than later. The `median_per_id` merge is the specific behaviour to fix first; the rest is API design.
